projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -482,35 +484,41 @@ private function get_inner_blocks( $inner_blocks ) {
 	/**
 	 * Gets the IDs of additional blocks on the Cart/Checkout pages or templates.
 	 *
-	 * @return void
+	 * @param string $cart_or_checkout Whether to get blocks on the cart or checkout page.
+	 * @return array All inner blocks on the page.
 	 */
-	public function get_additional_blocks_on_page() {
-		global $post;
-		$content = $post->post_content;
+	public function get_additional_blocks_on_page( $cart_or_checkout = 'cart' ) {
 
-		$cart_checkout_templates_in_use = class_exists( 'Automattic\WooCommerce\Blocks\Package' ) && version_compare( Automattic\WooCommerce\Blocks\Package::get_version(), '10.6.0', '>=' );
+		$additional_blocks_on_page_transient_name = 'jetpack_woocommerce_analytics_additional_blocks_on_' . $cart_or_checkout . '_page';
+		$additional_blocks_on_page                = get_transient( $additional_blocks_on_page_transient_name );
 
-		if ( $cart_checkout_templates_in_use ) {
-			if ( function_exists( 'get_block_template' ) ) {
-				$content = get_block_template( 'woocommerce/woocommerce//page-checkout' )->content;
-			}
+		if ( false !== $additional_blocks_on_page ) {
+			return $additional_blocks_on_page;
+		}
 
-			if ( function_exists( 'gutenberg_get_block_template' ) ) {
-				$content = gutenberg_get_block_template( 'woocommerce/woocommerce//page-checkout' )->content;
-			}
+		$content = $this->cart_content_source;
+
+		if ( 'checkout' === $cart_or_checkout ) {
+			$content = $this->checkout_content_source;
 		}
 
 		$parsed_blocks = parse_blocks( $content );
 		$other_blocks  = array_filter(
 			$parsed_blocks,
-			function ( $block ) {
+			function ( $block ) use ( $cart_or_checkout ) {
 				if ( ! isset( $block['blockName'] ) ) {
 					return false;
 				}
-				if ( is_checkout() && 'woocommerce/checkout' !== $block['blockName'] ) {
+				if ( 'woocommerce/classic-shortcode' === $block['blockName'] ) {
+					return false;
+				}
+				if ( 'core/shortcode' === $block['blockName'] ) {
+					return false;
+				}
+				if ( 'checkout' === $cart_or_checkout && 'woocommerce/checkout' !== $block['blockName'] ) {
```

Comment:
Nitpic: Similar as before. I wonder if we should add some line breaks to increase the readability. There's also an existing part below this function, which cannot be reviewed/commented, but could benefit from adding a line break.

```suggestion
				if ( 'woocommerce/classic-shortcode' === $block['blockName'] ) {
					return false;
				}
				
				if ( 'core/shortcode' === $block['blockName'] ) {
					return false;
				}
				
				if ( 'checkout' === $cart_or_checkout && 'woocommerce/checkout' !== $block['blockName'] ) {
```